core/internal/leet/livestore_test.go

```diff
@@ -0,0 +1,465 @@
+		}
+	}
+
+	// Next read should return EOF
+	_, err = ls.Read()
+	if !errors.Is(err, io.EOF) {
+		t.Fatalf("Expected EOF, got: %v", err)
+	}
+}
+
+// TestLiveStore_CloseIdempotent tests that Close can be called multiple times
+func TestLiveStore_CloseIdempotent(t *testing.T) {
+	tmpFile, err := os.CreateTemp(t.TempDir(), "close-*.wandb")
+	if err != nil {
+		t.Fatalf("CreateTemp: %v", err)
+	}
+	tmpFile.Close()
+	_ = os.Remove(tmpFile.Name())
+
+	// Write a valid header to a tranaction log.
+	w, err := transactionlog.OpenWriter(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("Failed to open transaction log for writing: %v", err)
+	}
+	w.Close()
+
+	ls, err := leet.NewLiveStore(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("NewLiveStore failed: %v", err)
+	}
+
+	// Close multiple times
+	if err := ls.Close(); err != nil {
+		t.Fatalf("First Close() failed: %v", err)
+	}
+	if err := ls.Close(); err != nil {
+		t.Fatalf("Second Close() failed: %v", err)
+	}
+	if err := ls.Close(); err != nil {
+		t.Fatalf("Third Close() failed: %v", err)
+	}
+}
+
+// TestLiveStore_ReadAfterClose tests that reading after close returns an error
+func TestLiveStore_ReadAfterClose(t *testing.T) {
+	tmpFile, err := os.CreateTemp(t.TempDir(), "read-after-close-*.wandb")
+	if err != nil {
+		t.Fatalf("CreateTemp: %v", err)
+	}
+	tmpFile.Close()
+	_ = os.Remove(tmpFile.Name())
+
+	// Create valid file with a record.
+	w, err := transactionlog.OpenWriter(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("Failed to open transaction log for writing: %v", err)
+	}
+	_ = w.Write(&spb.Record{Num: 1, Uuid: "test"})
+	w.Close()
+
+	ls, err := leet.NewLiveStore(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("NewLiveStore failed: %v", err)
+	}
+
+	// Close the LiveStore
+	ls.Close()
+
+	// Try to read after close
+	_, err = ls.Read()
+	if err == nil {
+		t.Fatal("Expected error when reading from closed LiveStore")
+	}
+	if !errors.Is(err, os.ErrClosed) && err.Error() != "livestore: db is closed" {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+}
+
+// TestLiveStore_LiveRead_ConcurrentWriterFlushes writes records in one goroutine
+// using the low-level LevelDB writer (so we can Flush) and reads them from another.
+// It ensures we see newly flushed records in order, and only get io.EOF between flushes.
+func TestLiveStore_LiveRead_ConcurrentWriterFlushes(t *testing.T) {
+	tmp, err := os.CreateTemp(t.TempDir(), "live-*.wandb")
+	if err != nil {
+		t.Fatalf("CreateTemp: %v", err)
+	}
+	_ = tmp.Close()
+
+	const total = 50
+
+	// Writer goroutine.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(path string) {
+		defer wg.Done()
+
+		f, err := os.OpenFile(path, os.O_WRONLY|os.O_TRUNC, 0o644)
+		if err != nil {
+			t.Errorf("open for write: %v", err)
+			return
+		}
+		defer f.Close()
+
+		// Match LiveStore (CRCAlgoIEEE, version 0).
+		w := leveldb.NewWriterExt(f, leveldb.CRCAlgoIEEE, 0)
+
+		for i := range total {
+			rec := &spb.Record{
+				Num:  int64(i),
+				Uuid: fmt.Sprintf("uuid-%d", i),
```

Comment:
Is this needed? I want to remove the `Uuid` field because it was only used in legacy code.